### PR TITLE
docs: readme

### PR DESCRIPTION
**Pull request by meniRoy**
I updated the README file
I will add the unit test section after unit test pr will merge.

### CODE REVIEW

**Comment by meniRoy** [comment]
[Add Docs to the README file](https://trello.com/c/WqnvrtGU/10056-add-docs-to-the-readme-file)

**Review by liranfarage89 — APPROVED** [review]
@meniRoy the README looks cool 🆒  and 💧 clear. Great job! 👏

File: README.md [hunk 1]
```diff
@@ -1,57 +1,96 @@
...
-* automatically filters environments that are associated with the branch you're currently working on
-* deploy, destroy, cancel, abort environments
-* get notifications about environments' changing statuses, and when waiting for approval
-* get live deployment log field in your VS Code terminal
+### 🚀 Execute Actions
+- Perform various actions such as `Deploy`, `Destroy`, `Cancel`, and `Abort` without leaving your IDE.
 
-## Requirements
+### 📜 Real-time Logs
+- Once an environment is selected, view the most recent deployment logs in the env0 output channel.
```
Comment (Wassap124) [inline README.md:14]:
```suggestion
- View the most recent deployment logs in the env0 output channel.
```

File: README.md [hunk 2]
```diff
@@ -1,57 +1,96 @@
...
-1. CMD+SHIFT+P
-2. search `reload`
-3. hit enter
+🔗 Learn how to [generate API keys](https://docs.env0.com/docs/api-keys).
 
-*Note* - some changes require to fully close the Debugging IDE (we've mainly encountered that when adding icons and such)
+## 🔑 Login and Logout
 
-Checking out the reading sources below is highly recommended
+To login or logout, simply use the `env0.login` and `env0.logout` commands through the VS Code command palette.
```
Comment (Wassap124) [inline README.md:31]: I would include the shortcut in parentheses here

File: README.md [hunk 3]
```diff
@@ -1,57 +1,96 @@
...
 
-## Extension Settings
+## 🚧 Limitations
 
-This extension contributes the following settings:
+**API Key & Organizations**: If you're using a personal API key and your account is associated with multiple organizations, the extension will default to a single organization and won't provide an option to select others. Use an Organization API key to resolve this.
```
Comment (Wassap124) [inline README.md:35]: move `Use an Organization API key to resolve this.` to a separate line just so it'd be more visible

File: README.md [hunk 4]
```diff
@@ -1,57 +1,96 @@
...
 
-* `env0.apiKeyId`
-* `env0.secretKey`
+## 🛠 Setting Up Your Development Environment
```
Comment (Wassap124) [inline README.md:37]: I think this should be changed to clarify these are tips / guidelines for the development environment - for developing THIS repo.

At first glance this seems like how to configure your env0 environment with the extension

File: README.md [hunk 5]
```diff
@@ -1,57 +1,96 @@
...
+Execute the following command in your terminal to run the integration tests:
+```bash
+pnpm run test
+```
+
+#### Debugging
+
+For debugging, go to the "Run and Debug" panel in VS Code and select "Extension Test."
+
+#### Technologies Used
```
Comment (Wassap124) [inline README.md:71]:
```suggestion
#### Libraries Used
```

**Review by Wassap124 — COMMENTED** [review]
good job! left some minor comments

**Comment by weinguy-env0** [comment]
I feel we should add some pictures and gifs of common views and usage. I can say for myself I usually tend to be wary of text only extensions.
@meniRoy how tough will it be? Or maybe you already added and it's outside this PR?

**Comment by weinguy-env0** [comment]
I would also go a bit easier on the emojis. A good example for an introduction gif + clean text for the marketplace page is this
https://github.com/mhutchie/vscode-git-graph
I'm not saying that we should remove all of the emojis, but I think cleaning this a bit will look better

**Comment by weinguy-env0** [comment]
Biggest problem I had was not finding my environment because the template I used didn't have a branch set. I think it's a thing we should mention to avoid user error. 
Another option would be in the hint in vscode when not finding any environments to add a link or possible explanation.

**Review by weinguy-env0 — COMMENTED** [review]
Left general comments

File: README.md (hunk 1 above)
Comment (meniRoy) [reply README.md]: d9df97038a544ab48e0cdbdc16fadf5c36e9959c

File: README.md (hunk 4 above)
Comment (meniRoy) [reply README.md]: d635674f6d5e87e51051ad9e9af9181e8bfb4ab6

File: README.md [hunk 6]
```diff
@@ -1,57 +1,96 @@
...
-4. on the `RUN AND DEBUG` select-box, choose `Run Extension`
-5. if there's a dialog with a warning, click on `Debug Anyway`
 
-By now, a new VS Code window should open up.  
-This is the window where you'll interact with your extension.  
-Whilst in the new window, open a git repo, which has any env0 environment associated with it  
+## 📋 Requirements
 
-*Note:* You will only see environments, in the organization of which you provided the api-keys   
+Before using this extension, make sure you meet the following prerequisites:
```
[the remark this answers is not in the document]
Comment (meniRoy) [reply README.md:22]: 69889c71687950601f6a9a457d54fc8c2d52fbe0

**Comment by meniRoy** [comment]
> Biggest problem I had was not finding my environment because the template I used didn't have a branch set. I think it's a thing we should mention to avoid user error. Another option would be in the hint in vscode when not finding any environments to add a link or a possible explanation.

Unfortunately, I cannot put links in the message section as discussed [here](https://env0.slack.com/archives/C041FQKHX2T/p1692810979393739)
69889c71687950601f6a9a457d54fc8c2d52fbe0
img99

File: README.md (hunk 2 above)
Comment (meniRoy) [reply README.md]: 1f679cf2904e67f5c1a829e6d341af4fe724f506

**Comment by meniRoy** [comment]
> I feel we should add some pictures and gifs of common views and usage. I can say for myself I usually tend to be wary of text only extensions. @meniRoy how tough will it be? Or maybe you already added and it's outside this PR?

I will add a GIF to this README along with the unit test section in a separate PR, after merging [this ticket](https://trello.com/c/ckpkkdTn/10785-fix-deployment-logs-synchronisation-with-deployment-status-update). Is that okay with you?

File: README.md (hunk 2 above)
Comment (Wassap124) [reply README.md]: `(Shift + Command + P / Ctrl + Shift + P)` -> `(Command / Ctrl + Shift + P)`

**Comment by weinguy-env0** [comment]
> > Biggest problem I had was not finding my environment because the template I used didn't have a branch set. I think it's a thing we should mention to avoid user error. Another option would be in the hint in vscode when not finding any environments to add a link or a possible explanation.
> 
> Unfortunately, I cannot put links in the message section as discussed [here](https://env0.slack.com/archives/C041FQKHX2T/p1692810979393739) [69889c7](https://github.com/env0/vscode-env0/commit/69889c71687950601f6a9a457d54fc8c2d52fbe0) img99

If not in VSCode please add a disclaimer in the readme

File: README.md (hunk 2 above)
Comment (meniRoy) [reply README.md]: 73f88c22a459bc8d67ebe18f32cded9cec15dfd2

**Review by weinguy-env0 — APPROVED** [review]

**Pull request merged**